### PR TITLE
Js files finder

### PR DESCRIPTION
**Pull request by nworr**
Refacto of code of lizmap controller to a dedicated class : reusable , testable.

js files are in several directory to check multiple case (default, js in project directory , subsfolder)  
Funded by 3liz

### CODE REVIEW

File: lizmap/modules/lizmap/lib/Project/UserFilesFinder.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+                $items = array(
+                    // current repository
+                    'media/js/',
+                    // or root (js shared over repositories)
+                    '../media/js/',
+                );
+                foreach ($items as $item) {
+                    $jsPathRoot = realpath($repositoryPath.$item.$dir);
+                    if (is_dir($jsPathRoot)) {
```
Comment (Gustry) [inline lizmap/modules/lizmap/lib/Project/UserFilesFinder.php:37]:
```suggestion
                    if (!is_dir($jsPathRoot)) {
```

To keep code on the left side of the editor, I find it more convenient negate, and a `continue` statement, so then all the code under can be pulled left.

Just a suggestion, about code visibility.

**Comment by Gustry** [comment]
Nice addition !
We will add it after in the project page table ;-)

File: lizmap/modules/lizmap/lib/Project/UserFilesFinder.php (hunk 1 above)
Comment (nworr) [reply lizmap/modules/lizmap/lib/Project/UserFilesFinder.php:34]: i'm not very keen of using "continue" statement, let's other reviewers decide ?

Comment (mdouchin) [reply lizmap/modules/lizmap/lib/Project/UserFilesFinder.php:34]: +1 for using continue.
This provides more readable content and avoids indentation nightmares

File: lizmap/modules/lizmap/lib/Project/UserFilesFinder.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * @copyright 2024 3liz.com
+ *
+ * @see      3liz.com
+ *
+ * @license   MIT
+ */
+
+namespace Lizmap\Project;
+
+class UserFilesFinder
```
Comment (mdouchin) [inline lizmap/modules/lizmap/lib/Project/UserFilesFinder.php:14]: I would call it `projectFilesFinder` instead, even if in the future, list of files may depend on the authenticated user.

**Review by mdouchin — CHANGES_REQUESTED** [review]
+1 for this nice modification, minor comments

File: lizmap/modules/lizmap/lib/Project/UserFilesFinder.php (hunk 1 above)
Comment (Gustry) [reply lizmap/modules/lizmap/lib/Project/UserFilesFinder.php:34]: This was not a strong opinion, just a way of coding that I prefer better.

Imagine a very long `if` block ... inside a loop. Having a `continue` ensures me that there is nothing under this very long `if` block. It puts some kind of safeguards that nothing will be executed under this statement

Just my 2 cents ;-)

**Review by Gustry — APPROVED** [review]
Nice ;-)

**Comment by nworr** [comment]
backport 3.9 3.8 3.7 ? 3.6 ??

**Comment by Gustry** [comment]
Try them all :)
But I'm not sure the CI will be valid for 3.6. We can make a non-official release.

**Comment by Gustry** [comment]
Good to go @nworr ?

**Review by rldhont — APPROVED** [review]

**Comment by rldhont** [comment]
@mdouchin we need your feedback to merge

**Comment by Gustry** [comment]
> @mdouchin we need your feedback to merge

@nworr did what was requested, just pressing the "resolved" button is enough ;-)

I just did it. Ok for merging ?

**Comment by nworr** [comment]
Thinking this PR, I found out a method to find files for whole repository would be great.
So back to "draft"

**Pull request merged**